### PR TITLE
xcodes: 1.5.0 -> 1.6.0, move to `by name`, `with lib;` cleanup, RFC format

### LINKED ISSUE #352844: Update request: xcodes 1.5.0 → 1.6.0
- Package name: xcodes
- Latest released version: 1.6.0

- Current version on the unstable channel: 1.5.0
- Current version on the stable/release channel: 1.4.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers** 

@emilytrau @0x120581f 

-----

Note for maintainers: 

Apple introduced Secure Remote Password as part of their auth flow, and a few tools relying on the old auth method are now failing to authenticate, including `xcodes`.

More info about the issue here: https://github.com/XcodesOrg/XcodesApp/issues/630

The new version of `xcodes` (1.6.0) implements SRP and works again as intended.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by matteo-pacini**
Fixes #352844 .

Also:
- Moved everything into `pkgs/by-name` (whilst adopting the new implicit `apple-sdk` pattern)
- Cleaned up `with lib;` usage in `meta`
- Applied RFC format

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilazy** [comment]
This doesn’t build for me on `x86_64-darwin`; is that a known issue?

**Comment by matteo-pacini** [comment]
@emilazy not entirely sure, but I think it flown below the radar because Swift didn't build on Hydra for a long time, and so did the packages that depend on it?

https://hydra.nixos.org/job/nixpkgs/trunk/xcodes.x86_64-darwin#tabs-status

The error is the following:

```/nix/store/wgcbqi40ahn1ay6905v190x5mlv9xdq9-apple-sdk-13.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:9: note: while building module 'CoreServices' imported from /nix/store/wgcbqi40ahn1ay6905v190x5mlv9xdq9-apple-sdk-13.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:
#import <CoreServices/CoreServices.h>
        ^
/nix/store/wgcbqi40ahn1ay6905v190x5mlv9xdq9-apple-sdk-13.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/MachineExceptions.h:22:10: note: while building module '_Builtin_intrinsics' imported from /nix/store/wgcbqi40ahn1ay6905v190x5mlv9xdq9-apple-sdk-13.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/MachineExceptions.h:22:
#include <emmintrin.h>
         ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "immintrin.h"
        ^
/nix/store/d2jm2gdf5qmyfbqk579f7h65sbi497sr-clang-wrapper-16.0.6/resource-root/include/immintrin.h:31:10: note: in file included from /nix/store/d2jm2gdf5qmyfbqk579f7h65sbi497sr-clang-wrapper-16.0.6/resource-root/include/immintrin.h:31:
#include <emmintrin.h>
         ^
/nix/store/d2jm2gdf5qmyfbqk579f7h65sbi497sr-clang-wrapper-16.0.6/resource-root/include/emmintrin.h:45:9: error: _Float16 is not supported on this target
typedef _Float16 __m128h_u __attribute__((__vector_size__(16), __aligned__(1)));
```

Not familiar with the internals of Clang/LLVM, but it sounds like it's trying to use an unsupported type (`_Float16`) on `x86_64-darwin`.
Sounds like an issue at the clang wrapper level rather than this derivation 🤔 .

**Comment by emilazy** [comment]
Oh, the problem is that you have to use the Swift `stdenv`. I forget why.

**Comment by matteo-pacini** [comment]
@emilazy works, thanks!

File: pkgs/by-name/xc/xcodes/package.nix [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+  fetchFromGitHub,
+  swift,
+  swiftpm,
+  swiftpm2nix,
+  makeWrapper,
+  aria2,
 }:
 let
   generated = swiftpm2nix.helpers ./generated;
+  stdenv = swift.stdenv;
```
Comment (emilazy) [inline pkgs/by-name/xc/xcodes/package.nix:12]: I’m not sure if this is any different from `swiftPackages.stdenv` but I think the latter is canonical.

**Review by emilazy — APPROVED** [review]

File: pkgs/by-name/xc/xcodes/package.nix (hunk 1 above)
Comment (matteo-pacini) [reply pkgs/by-name/xc/xcodes/package.nix]: Searched on the repo, and you're right - it seems to be more widely adopted.
Just pushed this change, thanks!

**Pull request merged**